### PR TITLE
fix: [CDS-70364]: StepGroup auto dimension layout

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

https://github.com/harness/harness-core-ui/assets/100121280/634d43dc-e063-4aeb-97fd-8ce30a7fb8f8

Jira: 
https://harness.atlassian.net/browse/CDS-70364
https://harness.atlassian.net/browse/CDS-55714

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.module.scss [hunk 1]
```diff
@@ -150,7 +126,7 @@
   }
 
   .stepGroupBody {
-    padding: var(--spacing-large) var(--spacing-xxxlarge);
+    padding: var(--spacing-large) calc(var(--spacing-xxxlarge) * 1.5);
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.module.scss:129]: Is there any reason behind `1.5` or does the padding look the best when using `1.5`? If there's a reason, let's add a comment

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/MatrixStepNode/MatrixStepNode.tsx [hunk 2]
```diff
@@ -12,81 +12,24 @@ import { Color } from '@harness/design-system'
...
   const allowAdd = defaultTo(props.allowAdd, false)
   const { getString } = useStrings()
   const [showAdd, setVisibilityOfAdd] = React.useState(false)
   const [showAddLink, setShowAddLink] = React.useState(false)
   const [isNodeCollapsed, setNodeCollapsed] = React.useState(false)
-  const [layoutStyles, setLayoutStyles] = React.useState<LayoutStyles>({ height: 100, width: 70 })
   const [concurrentNodes, setConcurrentNodes] = React.useState<number>(MAX_ALLOWED_MATRIX_COLLAPSED_NODES)
   const [{ visibilityMap }] = useCollapsedNodeStore()
-
+  const graphRef = React.useRef<HTMLDivElement>(null)
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineDiagram/Nodes/MatrixStepNode/MatrixStepNode.tsx:32]: Is this needed? It is used at line 91 but looks like the assigned ref is not used anywhere

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.module.scss (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.module.scss]: Thanks, was dev-testing have removed

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/MatrixStepNode/MatrixStepNode.tsx (hunk 2 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineDiagram/Nodes/MatrixStepNode/MatrixStepNode.tsx]: Thanks, removed

**Review by rameshx — APPROVED** [review]

**Review by srdjan-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Comment by PratyushGar** [comment]
Coverage is failing due to pipeline studio. which is planned to be added along with pipeline studio upgrades

**Pull request merged**